### ISSUE TITLE
[SC2312 triggers for return in a condition

### ISSUE THREAD

**Issue by Kreyren** [koalaman/shellcheck#3042]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2312
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0
- [-] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
	* It covers it by suggesting `|| true` which doesn't seem like a sane management in this scenario
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit
	* shellcheck.net does not show this as an issue -> Assuming invalid submitting for legacy

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

In https://github.com/NiXium-org/NiXium/pull/118/commits/1771b162785098712f7d4a0423e91bde73758e64

```sh
# shellcheck disable=SC2154 # Variables provided to the environment by Nix
echo "$systemDevice" >/dev/null
echo "$systemDeviceBlock" >/dev/null
echo "$secretTsvetanPasswordPath" >/dev/null
echo "$secretTsvetanKeyPath" >/dev/null

# FIXME-QA(Krey): This should be a runtimeInput
die() { printf "FATAL: %s\n" "$2"; exit ;}

# We have to use `env PATH=$PATH` so that used commands are ensured to use the correct PATH to see the expected binaries
# shellcheck disable=SC2068 # We expect the splitting
esudo() { sudo env "PATH=$PATH" $@ ;}

[ -b "$systemDevice" ] || die 1 "Expected device was not found, refusing to install for safety"

ragenixTempDir="/run/agenix"
ragenixIdentity="$HOME/.ssh/id_ed25519"

# Manage Ragenix - The Rust implementation of age for managing secret files
	[ -L "$ragenixTempDir" ] || {
		[ -d "$ragenixTempDir.d/1" ] || sudo mkdir "$ragenixTempDir.d/1"
		ln -s "$ragenixTempDir" "$ragenixTempDir.d/1"
	}
	sudo chown -R "root:keys" "$ragenixTempDir"
	sudo chmod -R 400 "$ragenixTempDir"

	[ -s "$ragenixTempDir/tsvetan-disks-password" ] || esudo age --identity "$ragenixIdentity" --decrypt --output "$ragenixTempDir/tsvetan-disks-password" "$secretTsvetanPasswordPath"

	[ -s "$ragenixTempDir/tsvetan-ssh-ed25519-private" ] || esudo age --identity "$ragenixIdentity" --decrypt --output "$ragenixTempDir/tsvetan-ssh-ed25519-private" "$secretTsvetanKeyPath"

# For the system to be able to process the runtime it needs 4GB of swap, if it does not have it then create a swap file
	swapFile="/swapfile"
	# shellcheck disable=SC2312 # We don't re-use the commands to need separate declaration, FIXME suggest upstream to adjust this check
	[ "$(awk '/SwapTotal/ {print $2}' /proc/meminfo)" >= 4194304 ] || {
		swapon -s | grep -q "$swapFile" || esudo swapoff "$swapFile" # Deactivate the swapfile if it's already there so that we can resize it
		[ "$(stat -c%s "$swapFile")" >= 4194304 ] || esudo dd if=/dev/zero of="$swapFile" bs=1M count=4096 conv=notrunc # Resize the swap file to the desired size

		# Ensure correct permissions
		esudo chmod 600 "$swapFile"
		esudo chown root:root

		esudo mkswap "$swapFile" # Make swap
	}

	swapon -s | grep -q "$swapFile" || esudo swapon "$swapFile" # Activate the swapfile if it's not already

nix build '.#nixos-tsvetan-stable' # pre-build the image

esudo disko-install \
	--flake "$FLAKE_ROOT#nixos-tsvetan-stable" \
	--mode format \
	--debug \
	--disk system "$(realpath "$systemDeviceBlock")" \
	--extra-files "$ragenixTempDir/tsvetan-ssh-ed25519-private" /nix/persist/system/etc/ssh/ssh_host_ed25519_key

# FIXME(Krey): Flash u-boot, currently blocked by https://github.com/OLIMEX/DIY-LAPTOP/issues/73 (flashing it manually via SPI clamp and ch341a programmer atm)

# FIXME(Krey): Flash firmware for keyboard
# FIXME(Krey): Flash firmware for touchpad
```

Removing `# shellcheck disable=SC2312 # We don't re-use the commands to need separate declaration, FIXME suggest upstream to adjust this check` causes:

![image](https://github.com/user-attachments/assets/657621f6-3391-435a-9bc0-8ee3074c46cf)

#### Here's what shellcheck currently says:

Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore).shellcheck[SC2312](https://www.shellcheck.net/wiki/SC2312)

#### Here's what I wanted or expected to see:

Do not show if the return is used as part of condition and not re-used in the code.

**Comment by Kreyren** [comment]
shellcheck.net does not show this as an issue -> Assuming invalid submitting for legacy

**Comment by Kreyren** [comment]
Correction: Invalid as it will fail evaluation with `set -e`
